Fix url/href issue

TheNicholasNick opened the pull request:
Previously the jist icon was added as a sibling to the a tag. then when
clicked the url that was retrieved was the 1st a tag in the parent node.
This doesn't work as links could be in the same node and might have
clicked the 2nd jist icon...
So now jist icons are added to the a tag, and thus the parent node is
the same a tag the user wants and its href is used as the url.
fixes issue on extratorrent.cc, magnet links now work again (2nd link)

Commit:
Fix url/href issue

Previously the jist icon was added as a sibling to the a tag. then when
clicked the url that was retrieved was the 1st a tag in the parent node.
This doesn't work as links could be in the same node and might have
clicked the 2nd jist icon...
So now jist icons are added to the a tag, and thus the parent node is
the same a tag the user wants and its href is used as the url.
fixes issue on extratorrent.cc, magnet links now work again (2nd link)

SoreGums commented:
Just running as an unpacked extension for now.
----
Actually hope to re-jig the actual event handling so it works after page load, as I am adding magnet links to already rendered pages, and this extension doesn't work because it only adds event listeners on page load. Once I figure it out I'll send another pull request ;)

Commit:
Refactor Success/Fail on Upload
Detect 200 success response code to accomdate when API goes down or there is some other issue with connection

Commit:
Merge branch 'refactor-success-fail-on-upload'

TheNicholasNick commented:
fail - didn't branch properly and messed up this pull req - redoing.
